mozilla-services/consvc-shepherd
# [AE-598] Add medium rectangle ads to Preview page

## References

[AE-598](https://mozilla-hub.atlassian.net/browse/AE-598)
https://github.com/mozilla-services/mars/pull/263

## Problem Statement

We recently added a new `newtab_rectangle` placement to Unified API. This placement serves [Medium Rectangle](https://www.iab.com/wp-content/uploads/2019/04/IABNewAdPortfolio_LW_FixedSizeSpec.pdf) format ads for New Tab.

The Shepherd Preview page does not currently allow previewing ads for this placement.

## Proposed Changes

This updates the preview page to fetch ads for the `newtab_rectangle` placement, and display them. The styling for these ads is not finalized, and may need to be tweaked in the future.

It also renames `image_src` to `image_url` for SPOCs to match the Unified API response.

## Verification Steps

1. Go to Preview page
2. Select `Dev Unified API` Environment
3. A medium rectangle ad should show under Rectangles

## Visuals

![Screenshot 2024-10-09 at 10 00 20 AM](https://github.com/user-attachments/assets/dd95d0bf-e4b7-4d70-9ef2-e9dee9eeacc6)


[AE-598]: https://mozilla-hub.atlassian.net/browse/AE-598?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
